### ISSUE TITLE
Using VictoriaMetrics components with VPA

### ISSUE THREAD

**Issue by bootlegsolutions** [VictoriaMetrics/VictoriaMetrics#5767]
### Is your question request related to a specific component?

everything using `lib/memory`

### Describe the question in detail

My particular usecase is related to `vmagent`, but would affect all of components utilizing `lib/memory`. If i'm reading it right, memory is allocated once at launch, and kept same through whole runtime.

I'm trying to run `vmagent` in k8s environment with VPA enabled. However, due to initial high resources given by VPA (around 10G of memory), `vmagent` reads that amount, and sets 9G VPZ for its entire runtime - despite only consuming ~3G RSS memory. This makes VPA never drop below 9G requested. 

Is it possible to reread memory during runtime via f.e. SIGHUP or something like that? What would be the recommended way of handling VPA with vmagent?

### Troubleshooting docs

- [X] General - https://docs.victoriametrics.com/Troubleshooting.html
- [X] vmagent - https://docs.victoriametrics.com/vmagent.html#troubleshooting
- [X] vmalert - https://docs.victoriametrics.com/vmalert.html#troubleshooting

**Comment by hagen1778** [comment]
> If i'm reading it right, memory is allocated once at launch, and kept same through whole runtime.

Not exactly. The memory is not allocated at launch. VM components determine the max limit of mem usage, so they could dedicate % of it to various internal structures. The dedicated % is used as cap, it is not pre-allocated immediately.

When VPA changes the memory limit, does it restart the process?

**Comment by bootlegsolutions** [comment]
If using "Recreate" policy, then yes (this is what i'm currently using). I'll check how it does with "Auto".

**Comment by hagen1778** [comment]
> I might be looking at it from wrong side, and the issue could be with the VPA instead - not entirely sure.

From my understanding, there is nothing vmagent can do with its memory usage - it uses exactly how much it needs, it doesn't pre-allocate memory just in case.
But the initial memory spike for vmagent could be related to the issue fixed in the following [commit](https://github.com/VictoriaMetrics/VictoriaMetrics-enterprise/commit/93153ab184b7dcc9b478e849d2e6a8f12739c9b5). Could you try if memory usage on start remains that high with vmagent v1.97.1?
